release/raydp_tests/raydp_benchmark.py

```diff
@@ -0,0 +1,220 @@
+import ray
+import ray.autoscaler.sdk
+import glob
+import csv
+import pandas as pd
+import numpy as np
+from pyspark.sql import SparkSession
+import raydp
+import os.path
+
+import time
+
+
+def load_dataset(spark, nbytes, npartitions, sort, base):
+    num_bytes_per_partition = nbytes // npartitions
+    filenames = []
+
+    @ray.remote
+    def load(i):
+        filename = os.path.join(base, "df-{}-{}-{}.parquet".format("sort" if sort else "groupby", num_bytes_per_partition, i))
+        print("Partition file", filename)
+        if not os.path.exists(filename):
+            if sort:
+                nrows = num_bytes_per_partition // 8
+                print("Allocating dataset with {} rows".format(nrows))
+                dataset = pd.DataFrame(np.random.randint(0, np.iinfo(np.int64).max, size=(nrows, 1), dtype=np.int64), columns=['a'])
+            else:
+                nrows = num_bytes_per_partition // (8 * 2)
+                print("Allocating dataset with {} rows".format(nrows))
+                dataset = pd.DataFrame(np.random.randint(0, 100, size=(nrows, 2), dtype=np.int64), columns=['a', 'b'])
+            print("Done allocating")
+            dataset.to_parquet(filename, flavor="spark")
+            print("Done writing to disk")
+        return filename
+
+    for i in range(npartitions):
+        filenames.append(load.remote(i))
+    ray.wait(filenames, num_returns=len(filenames))
+
+    return spark.read.parquet(os.path.join(base, "df-{}-{}-*.parquet".format("sort" if sort else "groupby", num_bytes_per_partition)))
+
+
+def trial(spark, nbytes, n_partitions, sort, generate_only, base):
+    df = load_dataset(spark, nbytes, n_partitions, sort, base)
+
+    if generate_only:
+        return
+
+    times = []
+    start = time.time()
+    for i in range(10):
+        print("Trial {} start".format(i))
+        trial_start = time.time()
+
+        if sort:
+            df.sort("a").head(10)
+        else:
+            df.groupBy("b").avg("a").collect()
+
+        trial_end = time.time()
+        duration = trial_end - trial_start
+        times.append(duration)
+        print("Trial {} done after {}".format(i, duration))
+
+        if time.time() - start > 10800:
+            break
+    return times
+
+
+def num_alive_nodes():
+    n = 0
+    for node in ray.nodes():
+        if node["Alive"]:
+            n += 1
+    return n
+
+
+def scale_to(target):
+    while num_alive_nodes() != target:
```

Comment:
Just removing it? We will anyway use the fixed size clutser right?

Reply:
`raydp.init_spark()` is very eager and will fail if we don't wait until the cluster is fully scaled, at initialization time. This used to be necessary, and I _think_ it's still necessary, but I'll test it out. @wuisawesome you originally added this, do you think it's still necessary?

Reply:
Hmm not necessary anymore while running locally.